helpdesk_mgmt_stage_server_action/models/helpdesk_ticket.py

```diff
@@ -10,31 +10,22 @@ class HelpdeskTicket(models.Model):
     @api.model_create_multi
     def create(self, vals_list):
         records = super().create(vals_list)
-        for record in records:
-            action = record.stage_id.action_id
-            if action:
-                context = {
-                    "active_model": self._name,
-                    "active_ids": [record.id],
-                }
-                action.with_context(**context).run()
+        records._run_stage_server_action()
         return records
 
     def write(self, vals):
-        records = "stage_id" in vals and self.filtered(
-            lambda l: l.stage_id.id != vals.get("stage_id")
-        )
-        if records:
-            res = super().write(vals)
-            action = (
-                self.env["helpdesk.ticket.stage"].browse(vals["stage_id"]).action_id
-            )
-            if action:
-                context = {
-                    "active_model": self._name,
-                    "active_ids": records.ids,
-                }
-                action.with_context(**context).run()
-        else:
-            res = super().write(vals)
+        res = super().write(vals)
+        if "stage_id" in vals:
```

Comment:
The previous method checked if the stage actually changed:

```
records = "stage_id" in vals and self.filtered(
            lambda l: l.stage_id.id != vals.get("stage_id")
```

That seems to have been lost in the refactor.
The unit test checking for this (`test_helpdesk_ticket_write`) was removed, too

This is change in behavior that doesn't seem intentional

Reply:
I've returned the check for different stages on the tickets, but the test you mentioned is checking something else and seems useless. The test I've written is almost enough to check the server action is triggered to run.